packages/core/src/App/Containers/AccountSwitcherWallet/account-switcher-wallet.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { useHistory } from 'react-router';
+import { Button, Text, useOnClickOutside } from '@deriv/components';
+import { useWalletAccountsList } from '@deriv/hooks';
+import { routes } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { Localize, localize } from '@deriv/translations';
+import { AccountSwitcherWalletItem } from './account-switcher-wallet-item';
+import './account-switcher-wallet.scss';
+
+type TAccountSwitcherWalletProps = {
+    is_visible: boolean;
+    toggle: (value: boolean) => void;
+};
+
+export const AccountSwitcherWallet = observer(({ is_visible, toggle }: TAccountSwitcherWalletProps) => {
+    const {
+        client: { switchAccount },
+        traders_hub: { setTogglePlatformType },
+        ui: { is_dark_mode_on },
+    } = useStore();
+
+    const { data: wallets } = useWalletAccountsList();
+    const dtrade_account_wallets = React.useMemo(() => wallets?.filter(wallet => wallet.dtrade_loginid), [wallets]);
+
+    const history = useHistory();
+
+    const wrapper_ref = React.useRef<HTMLDivElement>(null);
+
+    const validateClickOutside = (event: MouseEvent) =>
+        is_visible && !(event.target as HTMLElement).classList.contains('acc-info');
+
+    const closeAccountsDialog = () => {
```

Comment:
wrap it with `React.useCallback`, because you'll pass it as props in component. We need this because `AccountSwitcherWalletItem` component will be wrapped by `observer` and it will work like `React.memo`

Reply:
updated.